osutil/build_id.go

```diff
@@ -0,0 +1,119 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"debug/elf"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+)
+
+// ErrNoBuildID is returned when an executable does not contain a Build-ID
+var ErrNoBuildID = errors.New("executable does not contain GNU Build-Id")
+
+// BuildID is an array of bytes that identify a given build of an executable.
+type BuildID []byte
+
+// String returns the familiar representation, namely "BuildID[...]=..."
+func (id BuildID) String() string {
+	switch len(id) {
+	case 0x14: // SHA1 note:
```

Comment:
0x10 would be MD5 or UUID (per the fedora doc), which in `file` is printed as `"md5/uuid"`; I think we should do the same